### DISCUSSION TITLE
PLUS connect a new OpenIGTLink compatible tracker

### DISCUSSION THREAD

**Discussion by Barneyhuuu** [PlusToolkit/PlusLib#975] [Q&A]
Hello!
We're using a new optical locator ( AimPosition ), which is OpenIGTLink compatible. We overwrote the example file "trackerServer. CXX" in the source code to transfer real-time location information through OpenIGTLinkIF module to the 3D Slicer and it went well. Three transforms were sent to Slicer correctly (ReferenceStar, Probe, Pointer).
![2022-08-11 185207](https://user-images.githubusercontent.com/93598282/184122347-26ec9452-d4a1-4e1b-99af-dff584c83e9d.png)
Unfortunately, I had  trouble in writing the configuration file, I have tried so many times so any advice will be wonderful！
I wrote the configuration file with help of example configuration: http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceOpenIGTLinkTracker.html
When I launched PlusServer, OpenIGTLink began to sent Transform from waiting state, no significant error was reported.
![2022-08-11 191042](https://user-images.githubusercontent.com/93598282/184124976-1bbc0d4c-b3e3-4fa4-a7f9-62dcc87d17a7.png)
Then I open Slicer and set a OpenIGTLink port and actived it, the video stream was imported, but there is no transforms in data list. It seemed that Plus didn't receive the transform message.
![2022-08-11 191141](https://user-images.githubusercontent.com/93598282/184125629-589f1d15-33dd-4b4a-bc20-3b8be60bb7cf.png)
I have some question: 1. OpenIGTLink sent three real-time  matrices, so the MessageType in config file should be TRANSFROM ? Or TDATA? 2. There is no PortName or in DataSources, will this cause PlusToolkit can't recognize the sent messages？
I upload the config file, full log file and OpenIGTLick cpp file if needed.
Any advice  will be appreciated! Thank you!
[File.zip](https://github.com/PlusToolkit/PlusLib/files/9308332/File.zip)

**Comment by Sunderlandkyl** [comment]
Duplicate of https://github.com/PlusToolkit/PlusLib/discussions/971
